plugins/editor/class.editor.plugin.php

```diff
@@ -725,6 +725,12 @@ public function postController_editorUpload_create($Sender, $Args = array()) {
             ];
             $validImage = !empty($imageType) && in_array($imageType, $validImageTypes);
 
+            $this->EventArguments['CategoryID'] = Gdn::request()->post('CategoryID');
```

Comment:
In addition to these, we also have `$discussionID` and `$absoluteFileDestination` available which may be beneficial to this hook.  Maybe pass `$absoluteFileDestination` by reference.